File: app/brokers/paper_broker.py
import uuid

from app.brokers.interfaces.base_broker import (
    BaseBroker
)

from app.core.logger import logger


class PaperBroker(BaseBroker):

    async def place_order(
        self,
        symbol,
        side,
        quantity,
        price
    ):

        order = {
            "order_id": str(uuid.uuid4()),
            "symbol": symbol,
            "side": side,
            "quantity": quantity,
            "price": price,
            "status": "FILLED"
        }

        logger.info(
            f"Paper Order Filled: {order}"
        )

        return order

    async def cancel_order(
        self,
        order_id
    ):

        logger.info(
            f"Paper Order Cancelled: "
            f"{order_id}"
        )

    async def get_positions(
        self
    ):

        return []

    async def get_balance(
        self
    ):

        return {
            "balance": 100000
        }

File: backend/src/trading/brokers/paperBroker.ts
import { randomUUID } from "crypto";
import { BaseBroker } from "./baseBroker";
import { logger } from "../../core/logger";

export class PaperBroker extends BaseBroker {
  async placeOrder(symbol: string, side: string, quantity: number, price: number): Promise<any> {
    const order = {
      order_id: randomUUID(),
      symbol,
      side,
      quantity,
      price,
      status: "FILLED",
    };

    logger.info(`Paper Order Filled: ${JSON.stringify(order)}`);
    return order;
  }

  async cancelOrder(orderId: string): Promise<void> {
    logger.info(`Paper Order Cancelled: ${orderId}`);
  }

  async getPositions(): Promise<any[]> {
    return [];
  }

  async getBalance(): Promise<{ balance: number }> {
    return {
      balance: 100000.0,
    };
  }
}

BaseBroker.register("paper", PaperBroker);
export default PaperBroker;
